packages/compile-common/src/types.ts

```diff
@@ -0,0 +1,66 @@
+import { ContractObject } from "@truffle/contract-schema/spec";
+
+export type Compilation = {
+  sourceIndexes: string[];
+  contracts: CompiledContract[];
+  compiler: {
+    name: string | undefined;
+    version: string | undefined;
+  };
+};
+
+export interface CompilerResult {
+  compilations: Compilation[];
+}
+
+export interface Bytecode {
+  bytes: string;
+  linkReferences: LinkReference[];
+}
+
+export interface LinkReference {
+  offsets: number[];
+  name: string | null; // this will be the contractName of the library or some other identifier
+  length: number;
+}
+
+export type CompiledContract = {
+  contractName: string;
+  sourcePath: string;
+  source: string;
+  sourceMap: string;
+  deployedSourceMap: string;
+  legacyAST: object;
+  ast: object;
+  abi: object[];
+  metadata: string;
+  bytecode: Bytecode;
+  deployedBytecode: Bytecode;
+  compiler: {
+    name: string;
+    version: string;
+  };
+  devdoc: object;
+  userdoc: object;
+  immutableReferences: object;
+  generatedSources: any;
+  deployedGeneratedSources: any;
+};
+
+export interface WorkflowCompileResult {
+  compilations: Compilation[];
+  contracts: CompiledContract[];
+}
+
+export type CompileFunction = ({
```

Comment:
We don't need this anymore - we should instead add `sources` and `sourcesWithDependencies` to `Compiler` interface